routing/result_interpretation.go

```diff
@@ -0,0 +1,338 @@
+package routing
+
+import (
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// Instantiate variables to allow taking a reference from the failure reason.
+var (
```

Comment:
do these need to be mutable?

Reply:
I intentionally made them mutable, so that I can take the reference of it. They are already defined as `const` in `channeldb`. Or is it possible to have an immutable (non-const) variable? One that can only be initialized and is final thereafter.

Reply:
i see now, you can't take a reference to a const. makes sense when i think about it, but have never came across it. TIL

> Or is it possible to have an immutable (non-const) variable? One that can only be initialized and is final thereafter.

not that i'm aware